src/text-editor-presenter.coffee

```diff
@@ -880,11 +983,34 @@ class TextEditorPresenter
     if decoration.isType('overlay')
       @updateOverlaysState()
 
+  decorationPropertiesDidChange: (decoration, event) ->
+    {oldProperties} = event
+    if decoration.isType('line') or decoration.isType('gutter')
+      @removeDecorationInfoFromLineDecorationCaches(
+        decoration.id,
+        oldProperties,
+        decoration.getMarker().getScreenRange())
+      @addToLineDecorationCaches(decoration, decoration.getMarker().getScreenRange())
+      if decoration.isType('line') or Decoration.isType(oldProperties, 'line')
+        @updateLinesState()
+      if decoration.isType('line-number') or Decoration.isType(oldProperties, 'line-number')
+        @updateLineNumbersState()
+      if (decoration.isType('gutter') and not decoration.isType('line-number')) or
+      (Decoration.isType(oldProperties, 'gutter') and not Decoration.isType(oldProperties, 'line-number'))
+        @updateCustomGutterDecorationState()
+    else if decoration.isType('overlay')
+      @updateOverlaysState()
```

Comment:
I'd like to push this to a follow-up, which will involve some reorganization of text-editor-presenter-spec.coffee. 
Adding to the complexity of that file as it is feels counterproductive.